src/content/docs/en/guides/backend/supabase.mdx

```diff
@@ -3,10 +3,806 @@ title: Supabase & Astro
+:::
+
+### Sign In
+
+#### Sign In with Email
+
+1. Go to `src/pages`
+2. Create a new file named `signin.astro`
+3. Add the following code to `signin.astro`:
+
+```astro
+---
+// src/pages/signin.astro
+import Layout from "../layouts/Layout.astro"
+import { getSessionData, setSessionData } from "../lib/auth"
+
+let accessToken = Astro.cookies.get("sb-access-token").value
+let refreshToken = Astro.cookies.get("sb-refresh-token").value
+
+if (accessToken && refreshToken) {
+    await setSessionData(accessToken, refreshToken)
+    return Astro.redirect("/profile")
+}
+
+console.log(await getSessionData())
+---
+
+<Layout title="Sign In">
+    <div class="flex justify-center items-center">
+        <div class="col-6 form-widget" aria-live="polite">
+            <h1 class="text-3xl text-gray-800">Sign In</h1>
+            <form
+                class="flex flex-col space-y-4 mt-8"
+                aria-live="polite"
+            >
+                <input
+                    class="border rounded px-3 py-2"
+                    aria-live="polite"
+                    name="email"
+                    type="email"
+                    placeholder="Email"
+                />
+                <input
+                    class="border rounded px-3 py-2"
+                    aria-live="polite"
+                    name="password"
+                    type="password"
+                    placeholder="Password"
+                />
+                <button
+                    class="block px-3 py-2 w-full text-white bg-gray-800 border rounded hover:border-2"
+                    aria-live="polite"
+                    type="submit"
+                >
+                    Sign In
+                </button>
+                <p class="text-center text-gray-800">Need an account? <a href="/signup">Sign Up</a></p>
+            </form>
+        </div>
+    </div>        
+</Layout>
+
+<script>
+    import { signIn, getSessionData } from "../lib/auth";
+
+    const form = document.querySelector("form");
+
+    form?.addEventListener("submit", async (e) => {
+        e.preventDefault();
+        const formData = new FormData(e.target as HTMLFormElement);
+        const creds = {
+            email: formData.get("email") as string,
+            password: formData.get("password") as string,
+        };
+
+        const signin = await signIn(creds.email, creds.password);
+        if (signin.error !== null) {
+            console.log(signin.error);
+            alert(signin.error.message);
+            window.location.href = "/signin";
+        } else {
+            const session = await getSessionData();
+
+            console.log(session);
+
+            const maxAge = 15 * 24 * 60 * 60;
+            if(session) {
+                document.cookie = `sb-access-token=${session.data.session?.access_token}; path=/; expires=${session.data.session?.expires_at}; max-age=${maxAge}; SameSite=Lax; secure`;
```

Comment:
Astro has utility functions to work with cookies. We could replace `document.cookie` with `Astro.cookies`